compiler/src/dotty/tools/dotc/core/OrderingConstraint.scala

```diff
@@ -271,12 +256,14 @@ class OrderingConstraint(private val boundsMap: ParamBounds,
 
   /** The bound type `tp` without clearly dependent parameters.
    *  A top or bottom type if type consists only of dependent parameters.
+   *  TODO: try to do without normalization? It would mean it is more efficient
+   *  to pull out full bounds from a constraint.
    *  @param isUpper   If true, `bound` is an upper bound, else a lower bound.
    */
   private def normalizedType(tp: Type, paramBuf: mutable.ListBuffer[TypeParamRef],
       isUpper: Boolean)(implicit ctx: Context): Type =
     stripParams(tp, paramBuf, isUpper)
-      .orElse(if (isUpper) defn.AnyType else defn.NothingType)
+      .orElse(if (isUpper) defn.AnyKindType else defn.NothingType)
```

Comment:
@AleksanderBG Does this change remove the need for https://github.com/lampepfl/dotty/commit/f94606a8490f724882628565926b8605c8a28def ?

Reply:
I was planning on testing that. I'm not completely sure how to make myself certain that now there's no need for f94606a - AFAIU, the above change will make the bounds of an unconstrained variable be `>: Nothing <: AnyKind`, but I'm not sure if it's not possible to still get a `<: Any` constraint, because other `TypeBounds` default to `>: Nothing <: Any` anyway.

Reply:
Ah right, those need to be fixed too, I have a wip branch wĥere I attempted that but I don't know when I'll be able to get back to it: https://github.com/dotty-staging/dotty/commits/wildcard-bounds